app/controllers/metrics_controller.rb

```diff
@@ -0,0 +1,77 @@
+class MetricsController < ApplicationController
+  before_action :authorize_prison_request
+
+  def index
+    @start_date = start_date_from_range
+
+    if all_time?
+      counts = all_time_counts
+    else
+      counts = weekly_counts(year: @start_date.year,
+                             week: @start_date.cweek)
+    end
+
+    @dataset = MetricsPresenter.new(*counts)
+  end
+
+  def weekly
+    @prison = Prison.find(params[:prison])
+    @date = 1.week.ago.to_date
+
+    counts = Counters::CountVisitsByPrisonAndCalendarWeek.
+             where(prison_name: @prison.name).
+             where(year: @date.year, week: @date.cweek).
+             fetch_and_format
+
+    overdue_counts = Overdue::CountOverdueVisitsByPrisonAndCalendarWeek.
+                     where(prison_name: @prison.name).
+                     where(year: @date.year, week: @date.cweek).
+                     ordered_counters
+
+    percentiles = Percentiles::DistributionByPrisonAndCalendarWeek.
+                  where(prison_name: @prison.name).
+                  where(year: @date.year, week: @date.cweek).
+                  ordered_counters
+
+    @dataset = MetricsPresenter.new(counts, overdue_counts, percentiles)
+  end
+
+private
+
+  def weekly_counts(year: nil, week: nil)
+    counts = Counters::CountVisitsByPrisonAndCalendarWeek.
+             where(year: year, week: week).fetch_and_format
+
+    overdue = Overdue::CountOverdueVisitsByPrisonAndCalendarWeek.
+              where(year: year, week: week).ordered_counters
```

Comment:
If it's working for you, then great. ordered_counters was mainly meant to be a method to feed fetch_and_format, which in mixed from the various support modules when it isn't being declared explicitly.